File: src/Questbook.py
from Questline import Questline

class Questbook:
    def __init__(self):
        self.questlines = {}
    
    def addQuestline(self, name):
        self.questlines[name] = Questline(name)
    
    def addQuest(self, desc, questline):
        self.questlines[questline].addQuest(desc)
    
    def getQuest(self, questline):
        return self.questlines[questline]

File: src/questbook.cpp
#include <string>
#include "questbook.h"

using namespace std;

void Questbook::addQuestline(string questline_id, string questline_name) {
    questlines[questline_id] = Questline(questline_id, questline_name);
}

void Questbook::addQuest(string questline_id, string quest_desc) {
    questlines[questline_id].add(Quest(quest_desc));
}   

Quest Questbook::getQuest(string questline_id, int idx) {
    return questlines[questline_id].get(idx);
}